src/Lachain.Core/Blockchain/VM/ExternalHandler.cs

```diff
@@ -116,6 +121,8 @@ public static void Handler_Env_WriteLog(int offset, int length)
                 throw new InvalidContractException("Bad call to call function");
             var address = addressBuffer.Take(20).ToArray().ToUInt160();
             var value = SafeCopyFromMemory(frame.Memory, valueOffset, 32)?.ToUInt256().ToMoney();
+            var msgValue = SafeCopyFromMemory(frame.Memory, valueOffset, 32)?.ToUInt256();
```

Comment:
There is no need in `msgValue` variable as it copies the value of `value` and it calls `SafeCopyFromMemory` the second time with the same args

Reply:
Updated the code in the latest commit. Still we had to keep `value` and `msgValue` both as the type of them are different... `Moeny` and `UInt256` respectively